static/app/views/preprod/components/visualizations/appSizeCategories.tsx

```diff
@@ -0,0 +1,138 @@
+import {useTheme} from '@emotion/react';
+import type {PieSeriesOption} from 'echarts';
+
+import BaseChart, {type TooltipOption} from 'sentry/components/charts/baseChart';
+import {formatBytesBase10} from 'sentry/utils/bytes/formatBytesBase10';
+import {APP_SIZE_CATEGORY_INFO} from 'sentry/views/preprod/components/visualizations/appSizeVizUtils';
+import {type TreemapResults, TreemapType} from 'sentry/views/preprod/types/appSizeTypes';
+
+interface AppSizeCategoriesProps {
+  sizeMode: 'install' | 'download';
+  treemapData: TreemapResults;
+}
+
+export function AppSizeCategories(props: AppSizeCategoriesProps) {
+  const theme = useTheme();
+  const {treemapData, sizeMode} = props;
+
+  function calculateCategorySizes(element: any): Record<string, number> {
+    const categorySizes: Record<string, number> = {};
+
+    function traverse(node: any) {
```

Comment:
I think technically this calculation might output the wrong result (since we don't always have 100% node coverage in a parent) and we may want to compute this in the backend using our `FileAnalysis` logic.

Reply:
I didn't even realize we had this, I'll use that